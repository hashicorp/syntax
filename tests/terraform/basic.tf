# SYNTAX TEST "source.terraform" "A small but typical result"

terraform {
# ^^^^^^^^^ source.terraform meta.block.terraform
#         ^ punctuation.section.block.begin.terraform
  required_providers {
# ^^^^^^^^^^^^^^^^^^ entity.name.label.terraform
#                    ^ punctuation.section.block.begin.terraform
    azurerm = {
#   ^^^^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#             ^ meta.braces.terraform punctuation.section.braces.begin.terraform
      source  = "hashicorp/azurerm"
#     ^^^^^^ meta.braces.terraform meta.mapping.key.terraform string.unquoted.terraform
#             ^ keyword.operator.terraform
#                ^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
      version = "~> 2.65"
#     ^^^^^^^ meta.braces.terraform meta.mapping.key.terraform string.unquoted.terraform
#             ^ keyword.operator.terraform
#                ^^^^^^^ string.quoted.double.terraform
    }
  }

  required_version = ">= 1.1.0"
# ^^^^^^^^^^^^^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#                  ^ keyword.operator.assignment.terraform
#                     ^^^^^^^^ string.quoted.double.terraform
}
# <- punctuation.section.block.end.terraform

provider "azurerm" {
# ^^^^^^^^ source.terraform meta.block.terraform
#         ^^^^^^^ string.quoted.double.terraform
#                  ^ punctuation.section.block.begin.terraform
  features {}
# ^^^^^^^^ entity.name.label.terraform
}
# <- punctuation.section.block.end.terraform

resource "azurerm_resource_group" "rg" {
# ^^^^^^^^ source.terraform meta.block.terraform
#        ^ punctuation.definition.string.begin.terraform
#         ^^^^^^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
#                               ^ punctuation.definition.string.end.terraform
#                                 ^ punctuation.definition.string.begin.terraform
#                                  ^^ string.quoted.double.terraform
#                                    ^ punctuation.definition.string.end.terraform
#                                      ^ punctuation.section.block.begin.terraform
  name     = "myTFResourceGroup"
# ^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#          ^ keyword.operator.assignment.terraform
#             ^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
  location = "westus2"
# ^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#          ^ keyword.operator.assignment.terraform
#             ^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
}
# <- punctuation.section.block.end.terraform
