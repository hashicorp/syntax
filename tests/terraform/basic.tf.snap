># SYNTAX TEST "source.terraform" "A small but typical result"
#^ source.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform comment.line.terraform
>
>terraform {
#^^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#         ^ source.terraform meta.block.terraform
#          ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
># ^^^^^^^^^ source.terraform meta.block.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#         ^ punctuation.section.block.begin.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>  required_providers {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#                    ^ source.terraform meta.block.terraform meta.block.terraform
#                     ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
># ^^^^^^^^^^^^^^^^^^ entity.name.label.terraform
#^ source.terraform meta.block.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform comment.line.terraform
>#                    ^ punctuation.section.block.begin.terraform
#^ source.terraform meta.block.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform comment.line.terraform
>    azurerm = {
#^^^^ source.terraform meta.block.terraform meta.block.terraform
#    ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform variable.declaration.terraform
#              ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.begin.terraform
>#   ^^^^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>#             ^ meta.braces.terraform punctuation.section.braces.begin.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>      source  = "hashicorp/azurerm"
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#      ^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform string.unquoted.terraform
#            ^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#              ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform keyword.operator.terraform
#               ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#                ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                 ^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform
#                                  ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>#     ^^^^^^ meta.braces.terraform meta.mapping.key.terraform string.unquoted.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>#             ^ keyword.operator.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>#                ^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>      version = "~> 2.65"
#^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#      ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform meta.mapping.key.terraform string.unquoted.terraform
#             ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#              ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform keyword.operator.terraform
#               ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#                ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                 ^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform
#                        ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
>#     ^^^^^^^ meta.braces.terraform meta.mapping.key.terraform string.unquoted.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>#             ^ keyword.operator.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>#                ^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform comment.line.terraform
>    }
#^^^^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform
#    ^ source.terraform meta.block.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.end.terraform
>  }
#^^ source.terraform meta.block.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>  required_version = ">= 1.1.0"
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform variable.declaration.terraform
#                   ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                    ^ source.terraform meta.block.terraform variable.declaration.terraform
#                     ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                      ^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                              ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
># ^^^^^^^^^^^^^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                  ^ keyword.operator.assignment.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                     ^^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
># <- punctuation.section.block.end.terraform
#^ source.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform comment.line.terraform
>
>provider "azurerm" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#          ^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                 ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                  ^ source.terraform meta.block.terraform
#                   ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
># ^^^^^^^^ source.terraform meta.block.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#         ^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                  ^ punctuation.section.block.begin.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>  features {}
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^ source.terraform meta.block.terraform meta.block.terraform entity.name.label.terraform
#          ^ source.terraform meta.block.terraform meta.block.terraform
#           ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.begin.terraform
#            ^ source.terraform meta.block.terraform meta.block.terraform punctuation.section.block.end.terraform
># ^^^^^^^^ entity.name.label.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
># <- punctuation.section.block.end.terraform
#^ source.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform comment.line.terraform
>
>resource "azurerm_resource_group" "rg" {
#^^^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#        ^ source.terraform meta.block.terraform
#         ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#          ^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                                ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                 ^ source.terraform meta.block.terraform
#                                  ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                                   ^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                                     ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                      ^ source.terraform meta.block.terraform
#                                       ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
># ^^^^^^^^ source.terraform meta.block.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#        ^ punctuation.definition.string.begin.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#         ^^^^^^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                               ^ punctuation.definition.string.end.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                                 ^ punctuation.definition.string.begin.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                                  ^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                                    ^ punctuation.definition.string.end.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#                                      ^ punctuation.section.block.begin.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>  name     = "myTFResourceGroup"
#^^ source.terraform meta.block.terraform
#  ^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#      ^^^^^ source.terraform meta.block.terraform variable.declaration.terraform
#           ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#            ^ source.terraform meta.block.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#              ^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                               ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
># ^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#          ^ keyword.operator.assignment.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#             ^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>  location = "westus2"
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#          ^ source.terraform meta.block.terraform variable.declaration.terraform
#           ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#            ^ source.terraform meta.block.terraform variable.declaration.terraform
#             ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#              ^^^^^^^ source.terraform meta.block.terraform string.quoted.double.terraform
#                     ^ source.terraform meta.block.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
># ^^^^ variable.declaration.terraform variable.other.readwrite.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#          ^ keyword.operator.assignment.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>#             ^^^^^^^^^^^^^^^^^ string.quoted.double.terraform
#^ source.terraform meta.block.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform comment.line.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
># <- punctuation.section.block.end.terraform
#^ source.terraform comment.line.terraform punctuation.definition.comment.terraform
# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.terraform comment.line.terraform
>